helix-core/src/auto_pairs.rs

```diff
@@ -74,36 +101,32 @@ fn handle_open(
     close_before: &str,
 ) -> Transaction {
     let mut end_ranges = SmallVec::with_capacity(selection.len());
-
     let mut offs = 0;
 
     let transaction = Transaction::change_by_selection(doc, selection, |start_range| {
-        let start_head = start_range.head;
-
-        let next = doc.get_char(start_head);
-        let end_head = start_head + offs + open.len_utf8();
-
-        let end_anchor = if start_range.is_empty() {
-            end_head
-        } else {
-            start_range.anchor + offs
-        };
-
-        end_ranges.push(Range::new(end_anchor, end_head));
+        let cursor = start_range.cursor(doc.slice(..));
+        let next_char = doc.get_char(cursor);
+        let mut len_inserted = 0;
 
-        match next {
+        let change = match next_char {
             Some(ch) if !close_before.contains(ch) => {
-                offs += open.len_utf8();
-                (start_head, start_head, Some(Tendril::from_char(open)))
+                len_inserted += open.len_utf8();
```

Comment:
This could just be = instead of +=, it confused me initially because it seemed like it's used across iterations

Reply:
done